### ISSUE TITLE
OpenID Provider Configuration End Point clarification

### ISSUE THREAD

**Issue by jakubvozarik** [ConsumerDataStandardsAustralia/standards-maintenance#206]
# Request For Clarification
Can you please clarify below questions for GET /.well-known/openid-configuration  endpoint:

1)Does a DH need to maintain URI versioning for this API?
2)In the response, a DH needs to send the below endpoint values (hosted by DH). 
Does a DH need to maintain URI versioning for these endpoints?
 
·         authorization_endpoint
·         token_endpoint
·         introspection_endpoint
·         revocation_endpoint
·         userinfo_endpoint

Thank you

**Comment by perlboy** [comment]
Everything in this query relates to the [OpenID Connect Discovery 1.0 specification](https://openid.net/specs/openid-connect-discovery-1_0.html). I would suggest you read this.
 
> # Request For Clarification
> Can you please clarify below questions for GET /.well-known/openid-configuration endpoint:
> 1)Does a DH need to maintain URI versioning for this API?

If a DH wants to version their authorisation server they can do so but they must also realise that the `issuer` [must match the concatenation](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig) of the discovery document. That is to say, if a DH versions the endpoint they are also versioning the keyset/realm/issuer used for signing.

With that said, OpenID Connect Discovery **does not** approach things in a _**versioned**_ way but rather in an _**advertised capabilities**_ way. Ie. If the IdP adds additional signing or encryption types these would be advertised as changes to the discovery document not a new version.

> 2)In the response, a DH needs to send the below endpoint values (hosted by DH).
> Does a DH need to maintain URI versioning for these endpoints?
> · authorization_endpoint
> · token_endpoint
> · introspection_endpoint
> · revocation_endpoint
> · userinfo_endpoint

As per above. If the Holder wants to version these endpoints they can do so. While the `issuer` base uri and configuration document path is rigidly enforced by the spec, the discovery document allows for advertising of these endpoints at the Holder's discretion (provided the issuer used during signing matches the original base uri).

Once again though, interaction with these endpoints is governed not by fairly poor path based versioning but by the client interacting with these endpoints (and the parameters associated with their dynamic registration).

In summary, while a Holder **may** choose to version these endpoints/documents they must consider the implications of this decision while remaining compliant and the **value** of doing this versioning seems to be limited (ie. I ask "Why would you need to?").

**Comment by CDR-API-Stream** [comment]
Hi @jakubvozarik, @perlboy has covered off these questions comprehensively. Is any further clarification required?

**Comment by CDR-API-Stream** [comment]
This issue has been answered and it is being closed accordingly.
